src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -4054,39 +4054,38 @@ void C2_MacroAssembler::masked_op(int ideal_opc, int mask_len, KRegister dst,
 
 #ifdef _LP64
 void C2_MacroAssembler::vector_mask_operation(int opc, Register dst, KRegister mask,
-                                              Register tmp, int masklen, int masksize,
-                                              int vec_enc) {
+                                              int masklen, int masksize, int vec_enc) {
+  assert(VM_Version::supports_bmi1(), "Strange hardware");
```

Comment:
Also need to check for supports_lzcnt().

Reply:
We have separate checks for all `popcnt`, `lzcnt` and `tzcnt`, do you think the function should verify or just let the assembler throw after?

Reply:
Good to have assert here in this function as well.